lookout/style/typos/benchmarks/typo_commits_report.py

```diff
@@ -74,18 +88,63 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
                      the data retrieval.
         :return: List of `Comment`-s with `TypoFix` in JSON format.
         """
-        return [generate_comment(
-            filename=typo_fix.path,
-            line=typo_fix.line_number,
-            text=json.dumps(typo_fix._asdict()),
-            confidence=100) for typo_fix in self.run(ptr_to, data_service)]
+        comments = []
+        id_to_fix = []
+        for typo_fix in self.run(ptr_from, data_service):
+            id_to_fix.append(typo_fix.identifier)
+            comments.append(generate_comment(
+                filename=typo_fix.path,
+                line=typo_fix.line_number,
+                text=json.dumps(typo_fix._asdict()),
+                confidence=100))
+        self._log.debug("Suggestion to fix %s.\nTypo %s in the set: %s", id_to_fix,
+                        self.config["analyze"]["wrong_id"],
+                        self.config["analyze"]["wrong_id"] in id_to_fix)
+        return comments
+
+    @classmethod
+    def train(cls, ptr: ReferencePointer, config: Mapping[str, Any], data_service: DataService,
+              **data) -> IdTyposModel:
+        """
+        Return empty model if check_all_identifiers is True.
+
+        It helps to speed up report generation. Such approach is not correct for original \
+        IdTyposAnalyzer but ok for Spy class.
+        """
+        if config.get("check_all_identifiers", True):
+            return IdTyposModel()
+        return super().train(ptr, config, data_service, **data)
+
+    @staticmethod
+    def _find_new_lines(prev_content: str, content: str) -> List[int]:
```

Comment:
You can make this a normal method and avoid the static `_find_new_lines_return_value` trickery.